Bio/PDB/PDBList.py

```diff
@@ -701,88 +702,166 @@ def get_seqres_file(self, savefile="pdb_seqres.txt"):
+        action="store_true",
+        help="write all assemblies for each PDB entry to local pdb tree",
+    )
+    mode.add_argument(
+        "-single", help="retrieve single structure - accepts a single PDB code"
+    )
+    mode.add_argument(
+        "-multi",
+        help="retrieve a set of structures - accepts a comma separated list of PDB codes",
+    )
+
+    # Add mututally exclusive arguments for format
+    format = parser.add_mutually_exclusive_group()
+    format.add_argument(
+        "-pdb", action="store_true", help="downloads structures in PDB format"
+    )
+    format.add_argument(
+        "-xml", action="store_true", help="downloads structures in PDBML (XML) format"
+    )
+    format.add_argument(
+        "-mmtf", action="store_true", help="downloads structures in mmtf format"
+    )
+
+    # Add final options
+    others = parser.add_argument_group()
+    others.add_argument(
+        "-d",
+        action="store_true",
+        help="a single directory will be used as pdb_path, not a tree",
+    )
+    others.add_argument(
+        "-o", action="store_true", help="overwrite existing structure files"
+    )
+    others.add_argument(
+        "-with-assemblies",
+        action="store_true",
+        help="downloads assemblies along with regular entries",
+    )
+
+    # Parse arguments
+    args = parser.parse_args()
+
+    # Set path
+    pdb_path = args.pdb_path
+
+    # Instantiate PDBList class
+    pl = PDBList(pdb=pdb_path)
+
+    # Set file format
+    if args.pdb:
+        file_format = "pdb"
+    elif args.xml:
+        file_format = "xml"
+    elif args.mmtf:
+        file_format = "mmtf"
+    else:
+        file_format = "mmCif"
 
-        elif sys.argv[1] == "assemb":
-            # get all assembly structures
+    # Set options
+    if (args.d) or (args.pdb_path == os.getcwd()):
+        pl.flat_tree = True
+    overwrite = args.o
+    with_assemblies = args.with_assemblies
+
+    # Run
+    if args.update:
+        # Update pdb
+        print("updating local PDB at " + pdb_path)
+        pl.update_pdb(file_format=file_format, with_assemblies=with_assemblies)
+
+    elif args.all:
+        # Get the entire PDB
+        pl.download_entire_pdb(file_format=file_format)
+        if with_assemblies:
+            # Get all assembly structures
             pl.download_all_assemblies(file_format=file_format)
 
-        elif len(sys.argv[1]) == 4 and sys.argv[1][0].isdigit():
-            pdb_code = sys.argv[1]
-            # get single PDB entry
+    elif args.obsol:
+        # Get all obsolete entries
+        pl.download_obsolete_entries(pdb_path, file_format=file_format)
+
+    elif args.assemb:
+        # Get all assembly structures
+        pl.download_all_assemblies(file_format=file_format)
+
+    elif args.single:
+        pdb_code = args.single
+        # Get single PDB entry
+        pl.retrieve_pdb_file(
+            pdb_code, pdir=pdb_path, file_format=file_format, overwrite=overwrite
+        )
```

Comment:
What do you think about reworking this method to:
`retrieve_pdb_file(pdb_code: list[str], pdir: str, file_format: str, overwrite: bool, with_assemblies: bool)`

Reply:
I've updated the handling of a  `-codes` argument at the end so that it iterates through a list rather than reworking the method itself. Maybe changing the method itself will come in the new module that @JoaoRodrigues has mentioned.